### PR TITLE
Refactor string conversion between RCT2 and UTF8

### PR DESCRIPTION
**Pull request by IntelOrca**
This is my first effort to try and clean up the code for converting strings between RCT2's format and UTF-8.

RCT2 uses code page 1252 for most languages, although because it has modified the code page slightly to include some Polish characters I have kept in our own conversion table for it. For all the other code pages I am now using Win32 API so that we do not need to have our own built in tables. They are still present for non-Windows platforms until we start using a library such as libicu.

I have also started to move some of the code to using `std::string` and `std::string_view`.

I will see if I can write some unit tests for this before I merge.

### CODE REVIEW

File: src/openrct2/localisation/convert.cpp [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+#include <limits>
+#include "../core/String.hpp"
+#include "../core/Util.hpp"
+#include "Localisation.h"
+
+/**
+ * Decodes an RCT2 string to a wide char string still in the original code page.
+ * An RCT2 string is a multi-byte string where every two-byte code point is preceeded with a byte value of 255.
+ */
+static std::wstring DecodeToWideChar(const std::string_view& src)
```
Comment (AaronVanGeffen) [inline src/openrct2/localisation/convert.cpp]: This file has a lot of overlap with its title-cased brother `Convert.cpp`. Is it an earlier copy included by accident?

**Comment by AaronVanGeffen** [comment]
Nice to see some work on this! Hopefully, we can get `libicu` integrated for other platforms soon, too.

**Comment by Gymnasiast** [comment]
Since I cannot test this well on my own, I will supply some test files to see if it's working well.

File: src/openrct2/core/String.cpp [hunk 2]
```diff
@@ -18,6 +18,12 @@
 #include <stdexcept>
 #include <vector>
 
+#ifdef _WIN32
+#define NOMINMAX
```
Comment (Gymnasiast) [inline src/openrct2/core/String.cpp]: This chokes Travis, because it's already defined. Needs wrapping in ifndef.

**Comment by Gymnasiast** [comment]
[Accént2.zip](https://github.com/OpenRCT2/OpenRCT2/files/1913325/Accent2.zip)

Above file should show up in the Scenario list as `Accént`, with its description reading `Accènt`.

**Comment by Gymnasiast** [comment]
If I just start this and leave the title screen (more precisely: the LL one) open, it will crash within a minute.

GDB output: https://pastebin.com/nukCpaFV

**Pull request merged**